lib/broadside/deploy.rb

```diff
@@ -4,9 +4,11 @@ class Deploy
 
     attr_reader :target
 
+    delegate :tag, to: :target
+
     def initialize(options = {})
       @target = Broadside.config.get_target_by_name!(options[:target])
-      @tag = options[:tag] || @target.tag
+      @target.tag = options[:tag] if options[:tag]
```

Comment:
yeah, I went back and forth on that. IDK:
* while the target object in memory has now diverged from the configuration version, it's not like we're in danger of writing back to the config the new, one-time value
* but yeah, now the two are different
* _not_ modifying the target and updating the assertion doesn't actually fix the problem, it just passes the assertion
* which means [this logic](https://github.com/lumoslabs/broadside/blob/master/lib/broadside/ecs/ecs_deploy.rb#L34-L39) would be more complex

AFAICT, the tradeoff is target coherence vs. code complexity